plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -379,17 +379,28 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
 		return quotas, nil
 	}
 
-	// if we have limited resources enabled for this resource, always calculate usage
 	inputObject := a.GetObject()
 
+	// Usage of some resources cannot be counted in isolation. For example, when
+	// the resource represents a number of unique references to external
+	// resource. In such a case an evaluator needs to process other objects in
+	// the same namespace which needs to be known.
+	if accessor, err := meta.Accessor(inputObject); namespace != "" && err == nil {
+		if accessor.GetNamespace() == "" {
+			accessor.SetNamespace(namespace)
+		}
+	}
+
+	// if we have limited resources enabled for this resource, always calculate usage
```

Comment:
this issue i have with the code now is that usage is calculated all the time.  we should only need to calculate use if there is a matching limited resource (which is an alpha feature at moment) or if there is an actually interesting covering quota.  can this be structured to ensure that usage is only calculated if the above constraints are satisfied?